crates/flightrepl/src/lib.rs

```diff
@@ -369,19 +379,22 @@ pub async fn get_records(
     };
     let sql_command_bytes = sql_command.as_any().encode_to_vec();
 
-    let request = add_api_key(
-        FlightDescriptor::new_cmd(sql_command_bytes).into_request(),
-        api_key,
-    );
+    let request = FlightDescriptor::new_cmd(sql_command_bytes).into_request();
+    let request = add_api_key(request, api_key)?;
 
     let mut flight_info = client.get_flight_info(request).await?.into_inner();
     let Some(endpoint) = flight_info.endpoint.pop() else {
-        return Err(FlightError::Tonic(Status::internal("No endpoint").into()));
+        return Err(FlightError::Tonic(Box::new(Status::internal(
+            "No endpoint returned from server. Verify server configuration.",
+        ))));
```

Comment:
```suggestion
        return Err(FlightError::Tonic(Status::internal(
            "No endpoint returned from server. Verify server configuration.",
        )));
```

I don't think the Box::new is required